<!DOCTYPE html>
<html lang="en">

<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=yes"> <!-- or no?? -->

<title>sap218 site | blogs</title> <!-- browser tab title -->

<meta name="author" content="Samantha C Pendleton">
<meta name="description" content="data scientist">
<meta name="keywords" content="science, data">

<!-- bootstrap -->
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css"> <!-- Latest compiled and minified CSS -->
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script> <!-- jQuery library -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.16.0/umd/popper.min.js"></script> <!-- Popper JS -->
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js"></script> <!-- Latest compiled JavaScript -->

<!-- Bootstrap Settings <link rel="canonical" href="https://getbootstrap.com/docs/4.4/examples/carousel/"> -->
<!-- Custom styles for this template <link href="css/carousel.css" rel="stylesheet"> -->

<!-- Icons -->
<link rel="sheet" href="https://use.fontawesome.com/releases/v5.7.0/css/all.css" integrity="sha384-lZN37f5QGtY3VHgisS14W3ExzMWZxybE1SJSEsQp9S+oqd12jhcu+A56Ebc1zFSJ" crossorigin="anonymous"> 
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.0/css/all.css" integrity="sha384-lZN37f5QGtY3VHgisS14W3ExzMWZxybE1SJSEsQp9S+oqd12jhcu+A56Ebc1zFSJ" crossorigin="anonymous">
<!-- https://fontawesome.com/icons?d=gallery -->





<!--<link rel="shortcut icon" href="images/icon/favicon.ico" type="image/x-icon"> OLD--> 
<!--<link rel="shortcut icon" href="images/icon/favicon.png" type="image/x-icon">-->

<link rel="shortcut icon" href="../images/icon/favicon_io/favicon.ico" type="image/x-icon"> 
<link rel="shortcut icon" href="../images/icon/favicon_io/favicon-16x16.png" type="image/x-icon">
<link rel="shortcut icon" href="../images/icon/favicon_io/favicon-32x32.png" type="image/x-icon">
<link rel="shortcut icon" href="../images/icon/favicon_io/apple-touch-icon.png" type="image/x-icon">
<link rel="shortcut icon" href="../images/icon/favicon_io/android-chrome-512x512.png" type="image/x-icon">
<link rel="shortcut icon" href="../images/icon/favicon_io/android-chrome-192x192.png" type="image/x-icon">





</head>







<!-- BODY START -->
<body data-spy="scroll" data-target=".navbar" data-offset="50" style="background-color:white"> <!-- whitesmoke https://htmlcolorcodes.com/color-names/ -->






<!-- START OF HEADER -->
<header>
<nav class="navbar navbar-expand-md justify-content-center navbar-light fixed-top" style="background-color:lavenderblush;"> <!-- bg-light // lavenderblush // lemonchiffon // lightseagreen -->

<button class="navbar-toggler toggler-example" type="button" data-toggle="collapse" data-target="#navbarSupportedContent1" aria-controls="navbarSupportedContent1" aria-expanded="false" aria-label="Toggle navigation">
<span class="dark-blue-text"><i class="fas fa-bars fa-1x"></i></span></button>


<div class="collapse navbar-collapse justify-content-center" id="navbarSupportedContent1">

  <ul class="navbar-nav"> <!-- mr-auto"> -->
      <li class="nav-item"><a class="nav-link" href="../index.html#home"><b>sap218  <i class="fas fa-sun"></i></b></a></li> 

      <li class="nav-item dropdown">
        <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><b>blogs</b></a>
        <div class="dropdown-menu" aria-labelledby="navbarDropdown">
          <a class="dropdown-item" href="../index.html#blogs">recent</a>
          <!--<div class="dropdown-divider"></div>-->
          <a class="dropdown-item" href="archived_blogs.html">archived</a>
        </div>
      </li>

      <li class="nav-item"><a class="nav-link" href="../index.html#academia"><b>academia</b></a></li>
      <li class="nav-item"><a class="nav-link" href="../index.html#contact"><b>contact</b></a></li>
  </ul>

</div>


</nav>
</header>
<!-- END OF HEADER -->






<div class="container-sm"> <!-- STARTING CONTENT -->

<h1 id="home"><small>.</small></h2>

<br />

<!-- ARCHIVED BLOGS -->
<h2><small>my bioinformatics experience</small></h2>

<p><strong>06/10/2017</strong></p>

<br />

<p>
Early August 2017, I was accepted onto the Masters course: Data Science at Aberystwyth University. I was simultaneously excited and nervous: in preparation I started teaching myself <code>Python</code>.
<br>
I saw the position for a Summer researcher in bioinformatics for an undergrad student and was interested, I thought it would be good experience for my Masters course and I could gain practical knowledge. I know somewhat the base of Bioinformatics: studying and analysing DNA sequences. I'm no expert in coding or specifically bioinformatics, but more than willing to learn & teach myself.
</p>

<p>
So bioinformatics is a wide range of topics: studying DNA, or biological components of a person, e.g. blood biomarkers. This specific project was looking into sequences. 
</p>

<blockquote class="blockquote">
<p>"An analysis of current state of the art software on nanopore metagenomic data"</p>
</blockquote>

<p>
In preparation I installed Linux Mint partition for my laptop - this benefits my research (installing packages and source code of software); plus may be useful for my Masters - I chose Mint as I've had experience with it.
</p>


<p>
<strong>Week 1</strong>
<br>
For my first day, my supervisor, Amanda Clare suggested I read some papers about the sort of research that was conducted from the research team who collected these sequences. The research was in a mine from South Wales, UK. DNA was extracted to study the Earth's subsurface, where there is no internet, to find bacteria. Two datasets exist: the research team said the second results are better quality as improved protocols were used for DNA extraction.
<br>
The DNA was sequenced using new Nanopore technology: DNA goes in, an electric current runs through, proteins send signals creating a squiggle. The output sequences are <code>ACGT</code> codes. <code>GC</code> content in sequences are said to be more stable and <code>AT</code> are lower quality, especially if repetitive (possible error).
</p>
<p>
The first tool to use was <code>Goldilocks</code>. I spent some time figuring out how to run, since this was the first <code>Python</code> package I had to install. <code>Goldilocks</code> plots were interesting: we were able to see some low AT codes. 
<br>
I also looked into <code>Poretools</code> and <code>SAMtools</code> for file converting. <code>Poretools</code> also created histograms, which I was able to use to recreate plots from the original paper, a personal success. 
</p>

<p>
My supervisor of the project and I went to watch the nanopore minION in action. That first week was emotionally draining: all these new terms I didn't understand, packages to install, and imposter syndrome. But I finally produced some plots and asked for help to understand them.
</p>

<p>
<strong>Week 2</strong>
<br>
I have found out that the longest reads (via <code>Poretools</code>) were potential errors and the quality of the data is low overall. The research team left the location after 50 minutes however continued running the data: the longest reads occurred after 50 minutes. However, the <code>T</code> heavy reads are not the same as the longest reads so I'll be conducting tests if <code>T</code> heavy were after 50 mins too.
</p>


<p>
<strong>Week 3</strong>
<br>
Interestingly, one <code>T</code> heavy read is within the 50 mins; when using <code>BLAST</code>, the query cover is low though results included fungus and bacteria, which was what the research team was looking for. On a side note: I used the Linux command uniq to see there are no duplicate reads. I once again attempted using <code>Poretools</code> to produce squiggles, this yet again had no results, however I tried this on the data set without the MUX (QC reads: not quality filtered) reads and the results were ever so slightly different (still no diagrams though) - I noticed there was an issue opened for <code>Poretools</code>, however apparently the fix is to avoid using the <code>PIP</code> installer.
</p>




<p>
<strong>Week 4</strong>
<br>
At the beginning of the week, I met with Amanda and she saw the report I started: we agreed to look more into a comparison of the datasets: BP_v1 & BP_v2 (v2 is the better quality data). After discussing with one of the researchers, Andre Soares, it was mentioned that the datasets were at different times of the year (BP_v1: Dec 2016, BP2: Apr 2017) and the reason for low quality data is because it was field work.
<br>
I spent most of the week working on the report, improving it and including the analysis the the version 1 data. I need to <code>BLAST</code> both datasets fully but to do so I need to be added to AU network cluster since these jobs will be too heavy loaded for my personal laptop.
</p>

<p>
I <code>BLAST</code> some random reads of both datasets: specifically the long reads which weren't <code>T</code> heavy, plus some that were shorter - data set 1 had nothing useful at all to report, however data set 2 has bacteria: specifically bacteria that thrives in 40+ degrees - which is unusual as the mine was 15-20 degrees.
<br>
My supervisor and I mentioned about presenting my work through a poster to an bioinformatics lab - I have translated the report into a poster.
</p>



<p>
<strong>Week 5</strong>
<br>
I was finally connected to the cluster and ran a <code>BLAST</code> on both datasets, the first set (BP_v1) had barely any results: scores (query cover/read lengths) were all low (less than 50) with random species that included peppers and piranha. However, BP_v2 results were much better, results in ~800 which we could observe there were different types of bacteria within the subsurface.
</p>

<p>
So I went into the lab with the creator of <code>Goldilocks</code>, Sam Nicholls, since they were doing some lab work for their PhD. I saw him run DNA through Gel and extraction, then to take some pictures under UV light, which I found out that degrades DNA quality so it needs to be quick - I also got the opportunity to watch the centrifuge (machine: vortex) run.
</p>



<p>
<strong>Week 6</strong>
<br>
So this is a final wrapping up of this summer research project: I am to <code>BLAST</code> the datasets against each other, and look into the tool, <code>Pavian</code>, to create hierarchy tree diagrams from the results. I blasted BP_v1 against BP_v2 and vise versa: no similarities. However, BP_v1 against BP_v1 & plus BP_v2 against BP_v2 had reads that were very similar (90%+).
<br>
I had minor issues with <code>Pavian</code> as it only works with particular tool outputs - so I ran out of time and was not able to create any diagrams.
</p>

<p>
To complete up the project, I presented my work at a BCS Mid Wales "Show & Tell" event. I only had 24 hours notice but I enjoyed creating and presenting! I also presented my poster to the bioinformatics lab team and had discussions about the results.
</p>


<p>
I would like to thank:
<br>
<a href="https://twitter.com/afcaber" class="btn btn-secondary">Amanda Clare</a> 
<a href="https://twitter.com/GeoMicroSoares" class="btn btn-secondary">Andre Soares</a> 
<a href="https://twitter.com/samstudio8" class="btn btn-secondary">Sam Nicholls</a> 
</p>


<p>
<strong>Some videos / links:</strong>
</p>
<p>
<a href="https://twitter.com/afcaber/status/898194865980403712" class="btn btn-secondary">Nanopore MinION</a> watching it live 
</p>
<p>
<a href="https://twitter.com/sap218/status/910070653419704320" class="btn btn-secondary">PCR prep</a> a time-lapse 
</p>
<p>
<a href="https://twitter.com/sap218/status/914106415664500736" class="btn btn-secondary">Me and the poster</a> at BSC Show & Tell 
</p>
<p>
<a href="https://www.biorxiv.org/content/10.1101/288969v1" class="btn btn-secondary">Report</a> on bioRxiv 
</p>
<p>
<a href="https://github.com/sap218/misc/blob/master/bioinformatics_poster.pdf" class="btn btn-secondary">Poster</a> final
</p>




<p>
Tools mentioned in order:
<br>
<a href="https://academic.oup.com/bioinformatics/article/32/13/2047/1743888" class="btn btn-secondary">Goldilocks</a> 

<a href="https://academic.oup.com/bioinformatics/article/30/23/3399/207172" class="btn btn-secondary">Poretools</a> 

<a href="https://academic.oup.com/bioinformatics/article/25/16/2078/204688" class="btn btn-secondary">SAMtools</a> 

<a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="btn btn-secondary">BLAST</a> 

<a href="https://www.biorxiv.org/content/10.1101/084715v1" class="btn btn-secondary">Pavian</a> 
</p>










<br />
<br />



</div> <!-- /.container --> <!-- END CONTENT -->



<!-- FOOTER -->
<footer class="container">
	<p class="float-right"><a href="#home">Back to top</a></p>
	<p>sap218 &middot; <a href="https://twitter.com/sap218"><img src="../images/social_media/twitter.png" alt="twitter" style="width:30px;height:30px;"></a> &middot; <a href="https://github.com/sap218"><img src="../images/social_media/github.png" alt="twitter" style="width:30px;height:30px;"></a></p>

</footer>


</body> <!-- END OF BODY -->
</html>
